[feature] #2158: Update `parity_scale_codec` (#2591)
hyperledger/iroha: CHANGELOG.rst

=========

2.0.0-pre.rc.7
---------

Features
~~~~~~~~

> - 2338 Add `cargo-all-features` instrumentation.

Reviewer: Please consider prefixing a number with `hyperledger#`. It enables linking to issues and PRs

![image](https://user-images.githubusercontent.com/49983831/182821921-f53e0fa2-b56b-4c2a-9989-5f968eea327d.png)

Author: Done that. We'll nee to be a bit careful with older changelog entries.

> - 2564 Kagami tool algorithm options.
> - 2490 Implement ffi_export for freestanding functions.
> - 1891 Validate trigger execution. #1891: Validate permission for event triggers
> - 1988 Derive macros for Identifiable, Eq, Hash, Ord. #1988, #2437: Derive macros for Identifiable, Eq, Hash, Ord
> - 2434 FFI bindgen library.
> - 2073 Prefer ConstString over String for types in blockchain. #2073: Prefer ConstString over String for data_model
> - 1889 Add domain-scoped triggers.
> - 2098 Block header queries. #2098: add block header queries
> - 2467 Add account grant subcommand into iroha_client_cli.
> - 2301 Add transaction's block hash when querying it.
> - 2454 Add a build script to the parity-scale-decoder tool.
> - 2061 Derive macro for filters.
> - 2228 Add Unauthorized variant to smartcontracts query error. #2228: Add Unauthorized query error variant

Reviewer 2: ```suggestion
- 2564 Add Kagami tool algorithm options.
- 2490 Implement ffi_export for freestanding functions.
- 1891 Validate trigger execution.
- 1988, 2437 Derive macros for Identifiable, Eq, Hash, Ord.
- 2434 Add FFI bindgen library.
- 2073 Prefer ConstString over String for types in blockchain.
- 1889 Add domain-scoped triggers.
- 2098 Add block header queries.
- 2467 Add account grant subcommand into iroha_client_cli.
- 2301 Add transaction's block hash when querying it.
- 2454 Add a build script to the parity-scale-decoder tool.
- 2061 Derive macro for filters.
- 2228 Add Unauthorized variant to smartcontracts query error.
```

Fixes
~~~~~

> - 2422 Hide private keys in configuration endpoint response.
> - Hyperledger#2492: Fixes not all triggers being executed that match an
>   event.
> - 2504 Fix failing tps benchmark.
> - 2477 Fix bug when permissions from roles weren't counted.
> - 2416 Fix lints on macOS arm.
> - 2457 Fix tests flakiness related to shut down on panic. #2457: Add shut down on panic configuration
> - 2473 parse rustc --version instead of RUSTUP_TOOLCHAIN.
> - 1480 Shut down on panic. #1480: Add panic hook to exit program on panic
> - 2376 Simplified Kura, no async, two files.

Reviewer 2: ```suggestion
- 2422 Hide private keys in configuration endpoint response.
- 2492: Fix not all triggers being executed that match an
  event.
- 2504 Fix failing tps benchmark.
- 2477 Fix bug when permissions from roles weren't counted.
- 2416 Fix lints on macOS arm.
- 2457 Fix tests flakiness related to shutting down on panic.
- 2473 Parse rustc --version instead of RUSTUP_TOOLCHAIN.
- 1480 Add panic hook to exit programs on panic.
- 2376 Simplify Kura: no async, two files.
```

Refactor
~~~~~~~~
- Ensure all feature combinations work.
- 2468 Remove debug supertrait from permission validators.
- 2419 Remove explicit `drop`s.
- 2253 Add `Registrable` trait to `data_model`

> - Impl `Origin` instead of `Identifiable` for the data events.
>  : impl `Origin` instead of `Identifiable` for the data events

Reviewer 2: ```suggestion
- Use Impl `Origin` instead of `Identifiable` for the data events.
```

Documentation
~~~~~~~~~~~~~

> - Flakyness report for Jul 30.

Reviewer 2: ```suggestion
- Update flakyness report for Jul 30.
```
